### ISSUE TITLE
Consider using Miniconda as the basis for Windows installations, rather than pure Python

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#3942]
When designing the `install_sct.bat` script for native Windows installations, I used Python's virtual environments, rather than Miniconda (which we had been using for our `install_sct` unix installations). 

My reasoning for this was that, at the time, we were working towards a pip-installable SCT, and so I wanted to keep the install script as light and "standard" as possible (so that the transition to `pip` would be as straightforward as possible).

However, in the following months, we've stopped actively pursing a pip-installable SCT due to a host of newly-identified blockers. And, in the meanwhile, the use of virtualenvs has caused a few minor bits of awkwardness:

- On one OS, we have a Python virtual environment, while on the others, we use Conda environments. Meaning that, documentation pages such as [Installing SCT for development](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Contributing%3A-Installing-SCT-for-development) have to account for both environment types, making the documentation less straightfoward.
- Upgrading Python versions requires a documentation update to instruct the user to change their available version of Python, rather than that being a knob that we ourselves can control via the `conda create` command. (See also: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3924.)
   - To add to this: What happens if the user changes their Python version? What happens if they've already installed a newer version of Python? Can we trust that the user will fiddle with their PATH to make SCT's chosen version of Python available on the command line?

So, perhaps instead of instructing the user to install Python, we can instruct them to install Miniconda, and then use `conda create` in the `install_sct.bat` script in much the same way we do for unix platforms. 

(This hasn't really been a big problem presently, but the more I think about it, the more I anticipate that it will become a problem in the future.)

**Comment by mguaypaq** [comment]
That all sounds very reasonable. If/when we finally decide to make SCT more pip-installable, we'll have to figure out how to do the migration away from Miniconda anyway, for non-Windows environments. In the meantime, we might as well be uniform in using Miniconda, especially on Windows where we have fewer testing resources and expertise.

**Comment by joshuacwnewton** [comment]
I have very good news on this front!

----

At first, I had tried manually installing Miniconda myself, and found that it uses a GUI installer. The GUI installer does a couple of problematic things: A) Has a user-configurable installation location B) Doesn't add `conda` to the `PATH` by default C) is NOT portable (i.e. it modifies the Windows registry, shows up on the "Programs and Features" menu, adds "Anaconda Prompt" start menu entries, etc.)

This is markedly different than the UNIX `.sh` installer, which can be quietly invoked via the command line and installed in "portable" mode. Working around these quirks would be _very_ difficult, since our Miniconda would have to play nicely with any existing/future Conda installations, and we'd have to implore the user not to touch the installed Miniconda, lest they break SCT.

----

Thankfully, I've discovered that the Windows installer _does_ have a "portable" install mode (https://github.com/conda/conda/issues/1977):

```
start /wait "" Miniconda3-latest-Windows-x86_64.exe /InstallationType=JustMe /AddToPath=0 /RegisterPython=0 /NoRegistry=1 /S /D=%cd%\python
```

So, not only do we get to replicate the behavior we currently use for `install_sct`, but we don't even need to list Miniconda as a prereq in our docs like we're doing with Python currently! We can install Miniconda ourselves by `curl`'ing the installer, then running it in the SCT dir! :tada:

Given this, I think it's _definitely_ in our best interest to use Miniconda for the Windows installer. 😊
